service/http.proxy/src/main/java/org/kaazing/gateway/service/http/proxy/HttpProxyServiceHandler.java

```diff
@@ -94,10 +95,31 @@ public void initializeSession(IoSession session, ConnectFuture future) {
             HttpConnectSession connectSession = (HttpConnectSession) session;
             connectSession.setVersion(acceptSession.getVersion());
             connectSession.setMethod(acceptSession.getMethod());
-            connectSession.setRequestURI(acceptSession.getRequestURI());
+            if (!validateRequestPath()) {
+                acceptSession.setStatus(CLIENT_NOT_FOUND);
+                acceptSession.close(false);
+                return;
+            }
+            String acceptPathInfo = acceptSession.getPathInfo().toString();
+            if (acceptPathInfo.indexOf("/") == 0) {
+                acceptPathInfo = acceptPathInfo.substring(1);
+            }
+            URI connectRequestURI = connectSession.getRequestURI().resolve(acceptPathInfo);
+            connectSession.setRequestURI(connectRequestURI);
             processRequestHeaders(acceptSession, connectSession);
         }
 
+        private boolean validateRequestPath() {
+            URI requestURI = acceptSession.getRequestURI();
+            String acceptPath = acceptSession.getServicePath().getPath();
+            String requestPath = requestURI.normalize().getPath();
+            if (requestPath.indexOf("/\\.\\./") == 0 || requestPath.indexOf(acceptPath) != 0) {
```

Comment:
I think the whole if condition can be eliminated (infact indexOf is wrong check as it checks anywhere in the path). Doesn't the following work ?
```
return requestPath.startsWith(acceptPath)
```